core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -210,6 +211,20 @@ private TableProperties() {
   public static final String SPARK_WRITE_PARTITIONED_FANOUT_ENABLED = "write.spark.fanout.enabled";
   public static final boolean SPARK_WRITE_PARTITIONED_FANOUT_ENABLED_DEFAULT = false;
 
+  public static final String PARTITIONED_FANOUT_WRITERS_CACHE_SIZE = "write.partition.fanout.writers-cache-size";
+  public static final int PARTITIONED_FANOUT_WRITERS_CACHE_SIZE_DEFAULT = Integer.MAX_VALUE;
+
+  public static final String PARTITIONED_FANOUT_WRITERS_CACHE_EVICT_MS =
+      "write.partition.fanout.writers-cache-eviction-ms";
+  public static final long PARTITIONED_FANOUT_WRITERS_CACHE_EVICT_MS_DEFAULT = TimeUnit.MINUTES.toMillis(5);
```

Comment:
I think the default value should not be 5 minutes. To keep the behavior consistent with before. Maybe we should the value never expire.

Reply:
Makes sense, I'll adjust the default value later

Reply:
What is the purpose of a timeout? I don't see how that aligns with the need to reduce memory consumption. You could create so many files in 5 minutes that you run out of memory. If you could evict based on memory consumption of the writers, that seems valuable. But I think I would remove this.

Reply:
@rdblue The reason for adding the timeout option is that it is often difficult to determine the memory consumption based on the number of writers